wire-ios-sync-engine/Source/UserSession/ZMUserSession+RecurringAction.swift

```diff
@@ -17,38 +17,55 @@
 //
 
 import Foundation
+import WireDataModel
 
 extension ZMUserSession {
 
-    func refreshUsersMissingMetadata(interval: TimeInterval = 3 * .oneHour) -> RecurringAction {
+    var refreshUsersMissingMetadataAction: RecurringAction {
+        .init(id: #function, interval: 3 * .oneHour) { [weak self] in
+
+            guard let moc = self?.managedObjectContext else { return }
+            moc.performGroupedAndWait { moc in
 
-        .init(id: "refreshUserMetadata", interval: interval) { [weak self] in
-            self?.perform {
                 let fetchRequest = ZMUser.sortedFetchRequest(with: ZMUser.predicateForUsersArePendingToRefreshMetadata())
-                guard let users = self?.managedObjectContext.fetchOrAssert(request: fetchRequest) as? [ZMUser] else {
+                guard let users = moc.fetchOrAssert(request: fetchRequest) as? [ZMUser] else {
                     return
                 }
+
                 users.forEach { $0.refreshData() }
+                moc.saveOrRollback()
             }
         }
-
     }
 
-    func refreshConversationsMissingMetadata(interval: TimeInterval = 3 * .oneHour) -> RecurringAction {
+    var refreshConversationsMissingMetadataAction: RecurringAction {
+        .init(id: #function, interval: 3 * .oneHour) { [weak self] in
+
+            guard let moc = self?.managedObjectContext else { return }
```

Comment:
**nitpick (non-blocking):** I think we usually call it `context` instead of `moc` across the code base in order to be consistent